bika/lims/browser/bika_listing.py

```diff
@@ -786,31 +786,31 @@ def _process_request(self):
         # and modify self.columns[]['toggle'] to match.
         toggle_cols = self.get_toggle_cols()
         for col in self.columns.keys():
-            if col in toggle_cols:
-                self.columns[col]['toggle'] = True
-            else:
-                self.columns[col]['toggle'] = False
+            self.columns[col]['toggle'] = col in toggle_cols
 
     def get_toggle_cols(self):
-        _form = self.request.form
-        toggles = {}
-        try:
-            default_toggles = _form.get("toggle_cols", "{}")
-            _key = self.form_id + "_toggle_cols"
-            form_toggles = _form.get(_key, default_toggles)
-            toggles = json.loads(form_toggles)
-        except (ValueError, TypeError):
-            pass
-        finally:
-            if not toggles:
-                toggles = {}
-        cookie_key = "%s%s" % (self.context.portal_type, self.form_id)
-        toggle_cols = toggles.get(
-            cookie_key, [col for col in self.columns.keys()
-                         if col in self.review_state['columns'] and
-                         ('toggle' not in self.columns[col] or
-                          self.columns[col]['toggle'] is True)])
-        return toggle_cols
+        """
+        Returns the list of column ids to be displayed for the current list.
+        :return: list of column ids to be displayed
+        :rtype: list of str
+        """
+        # Get the toggle configuration from the cookie
+        cookie_toggle = self.request.get('toggle_cols', '{}')
```

Comment:
Shouldn't be that value in `self.request.cookies`?

Reminder: We have to consolidate all the different places in the system where cookies get set, because these cookies make caching by Varnish impossible.